packages/block-library/src/group/edit.js

```diff
@@ -63,11 +66,31 @@ function GroupEditControls( { tagName, onSelectTagName } ) {
 					{ label: '<article>', value: 'article' },
 					{ label: '<aside>', value: 'aside' },
 					{ label: '<footer>', value: 'footer' },
+					{ label: '<nav>', value: 'nav' },
 				] }
 				value={ tagName }
-				onChange={ onSelectTagName }
+				onChange={ ( value ) => {
+					setAttributes( {
+						tagName: value,
+						ariaLabel: value === 'nav' ? ariaLabel : undefined,
+					} );
+				} }
 				help={ htmlElementMessages[ tagName ] }
 			/>
+			{ tagName === 'nav' && (
+				<TextControl
+					label={ __( 'Navigation label' ) }
+					value={ ariaLabel || '' }
+					__next40pxDefaultSize
+					__nextHasNoMarginBottom
+					onChange={ ( value ) => {
+						setAttributes( { ariaLabel: value } );
+					} }
+					help={ __(
+						'Add a label to describe the purpose of this navigation element.'
+					) }
+				/>
+			) }
```

Comment:
I wonder if it's just time to add a full standardized UI for the aria label block support 🤔 

In reality users can already set this attribute in code today. It just doesn't have a UI... I'm not sure only showing it for the `nav` tag name makes it more clear 🤔

Reply:
> I wonder if it's just time to add a full standardized UI for the aria label block support 

That's certainly true, and maybe we should do so in the future.

Personally, I'm cautious about changing the block support spec right now. For now, I prefer to expose the UI only to the Group block. I'd like to hear other people's opinions on this.